src/global/project/manifest.rs

```diff
@@ -468,6 +468,29 @@ impl FromStr for Mapping {
     }
 }
 
+#[derive(Default)]
+pub enum ExposedType {
+    #[default]
+    All,
+    Subset(Vec<Mapping>),
+}
+
+impl ExposedType {
+    pub fn from_mapping(mapping: Vec<Mapping>) -> Self {
+        match mapping.is_empty() {
+            true => Self::All,
+            false => Self::Subset(mapping),
+        }
+    }
+
+    pub fn from_bool(auto_exposed: bool) -> Self {
+        match auto_exposed {
+            true => Self::All,
+            false => Self::Subset(Default::default()),
+        }
+    }
```

Comment:
Maybe split that into two methods. One for creating the struct in the state where nothing is exposed, and one methods where everything is exposed?

Reply:
I could call directly default on it, let me try